src/mbed_tools/project/_internal/project_data.py

```diff
@@ -120,7 +122,7 @@ def from_existing(cls, root_path: Path) -> "MbedProgramFiles":
             mbed_os_ref=mbed_os_file,
             cmakelists_file=cmakelists_file,
             cmake_config_file=root_path / CMAKE_CONFIG_FILE_PATH,
-            cmake_build_dir=root_path / CMAKE_BUILD_DIR,
+            cmake_build_dir=root_path / CMAKE_BUILD_DIR / build_subdir,
```

Comment:
Isn't the CMAKE dir the childmost directory? This line makes it sound like the cmake directory has the profile-specific folders under it. Probably `CMAKE_BUILD_DIR` is just a misleading name. Consider updating it to something like `BUILD_DIR`, to avoid implying that the cmake build directory is that directory.